### DISCUSSION TITLE
Macro recording to file

### DISCUSSION THREAD

**Discussion by danriedl** [helix-editor/helix#9367] [Ideas]
I would appreciate a feature that allows me to record a macro to a file. 
This file could be stored in a designated directory, such as `.config/helix/macros/`. 

Later, I would like the ability to load the recorded macro into the register using a macro_picker. 
Additionally, it would be great if there was an option to open a file with the command `hx -s .config/helix/macros/join_all_lines.hx mytextfile.txt` to apply the recorded macro directly to the file.

**Comment by danriedl** [comment]
As macros are recorded to the `@` register as string, this could also be passed to the binary.

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/pull/4767#issuecomment-1316102743
